# Tarea 1 Ciencia de redes

Juntense en grupos de no más de 4 personas. Tienen una semana para entregar esta tarea. Para ambos datasets `example_1.txt` y `example_2.txt`. Cada grupo deberá cargar el edge list que corresponde y hacer lo siguiente:

* Pregunta 1: Calcula todas las medidas básicas que se muestran arriba. Comente ¿Qué se puede sospechar acerca de la distribución de grados de la red solo en función de la media y los grados extremos?

* Pregunta 2: Graficar la distribución del grado en escala log-log. También graficarlo en escala lineal. Comente cómo encaja esto con el análisis de la pregunta 1.

* Pregunta 3: Dibuje la red utilizando los dos algoritmos de diseño que se muestran arriba. ¿Cómo se refljan en la apariencia de la red los hallazgos de las preguntas 1 y 2?

*Nota: Recuerden que SU interpretación de los resultados es lo más importante.*

In [1]:
import pandas as pd 
import numpy as np 
import networkx as nx
import matplotlib.pyplot as plt

Cargamos redes

In [4]:
G1 = nx.read_edgelist('./example_1.txt', comments='#',
                     create_using=nx.Graph(), 
                     delimiter=' ', 
                     nodetype=int, 
                     encoding='utf-8')

G2 = nx.read_edgelist('./example_2.txt', comments='#',
                     create_using=nx.Graph(), 
                     delimiter=' ', 
                     nodetype=int, 
                     encoding='utf-8')

## Pregunta 1

### Calcula todas las medidas básicas que se muestran arriba. Comente ¿Qué se puede sospechar acerca de la distribución de grados de la red solo en función de la media y los grados extremos?

Dataset 1

In [17]:
N = len(G1)
L = G1.size()
degrees = list(dict(G1.degree()).values())
kmin = min(degrees)
kmax = max(degrees)
media = np.mean(degrees)
print(f"La red tiene en un grado promedio de {media}, con un grado mímino de {kmin} y un grado máximo de {kmax}")

La red tiene en un grado promedio de 5.982, con un grado mímino de 3 y un grado máximo de 98


R: Tanto la mediana (4) como el promedio (5.982) de los grados de la red es cercano a su grado mínimo (3), por lo que se puede sospechar que la distribución de grados de la red es asimétrica a la izquierda (asimétrica positiva) y con una cola extensa, pues el grado máximo es 98. 